aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255_bulletproofs.move

```diff
@@ -99,12 +106,69 @@ module aptos_std::ristretto255_bulletproofs {
+    /// `v` satisfy `v` in `[0, 2^num_bits)`.
+    /// Only works for `num_bits` in `{8, 16, 32, 64}` and batch size (length of `comms`) in `{1, 2, 4, 8, 16}`.
+    public fun verify_batch_range_proof_pedersen(
+        comms: &vector<pedersen::Commitment>, proof: &RangeProof,
+        num_bits: u64, dst: vector<u8>): bool
+    {
+        assert!(features::bulletproofs_batch_enabled(), error::invalid_state(E_NATIVE_FUN_NOT_AVAILABLE));
+
+        let comms = std::vector::map_ref(comms, |com| ristretto255::point_to_bytes(&pedersen::commitment_as_compressed_point(com)));
+
+        verify_batch_range_proof_internal(
+            comms,
+            &ristretto255::basepoint(), &ristretto255::hash_to_point_base(),
+            proof.bytes, num_bits, dst
+        )
+    }
+
+    // Verifies a zero-knowledge range proof for a batch of commitments `comms` (each of the form 
+    /// `v * val_base + r * rand_base`), ensuring that all values `v` satisfy 
+    /// `v` in `[0, 2^num_bits)`. Only works for `num_bits` in `{8, 16, 32, 64}` and batch size 
+    /// (length of the `comms`) in `{1, 2, 4, 8, 16}`.
+    public fun verify_batch_range_proof(
+        comms: &vector<pedersen::Commitment>,
+        val_base: &RistrettoPoint, rand_base: &RistrettoPoint,
+        proof: &RangeProof, num_bits: u64, dst: vector<u8>): bool
+    {
+        assert!(features::bulletproofs_batch_enabled(), error::invalid_state(E_NATIVE_FUN_NOT_AVAILABLE));
+
+        let comms = std::vector::map_ref(comms, |com| ristretto255::point_to_bytes(&pedersen::commitment_as_compressed_point(com)));
+
+        verify_batch_range_proof_internal(
+            comms,
+            val_base, rand_base,
+            proof.bytes, num_bits, dst
+        )
+    }
+
     #[test_only]
     /// Computes a range proof for the Pedersen commitment to 'val' with randomness 'r', under the default Bulletproofs
     /// commitment key; see `pedersen::new_commitment_for_bulletproof`. Returns the said commitment too.
     ///  Only works for `num_bits` in `{8, 16, 32, 64}`.
     public fun prove_range_pedersen(val: &Scalar, r: &Scalar, num_bits: u64, dst: vector<u8>): (RangeProof, pedersen::Commitment) {
         let (bytes, compressed_comm) = prove_range_internal(scalar_to_bytes(val), scalar_to_bytes(r), num_bits, dst, &ristretto255::basepoint(), &ristretto255::hash_to_point_base());
+        let comm = pedersen::new_commitment_from_bytes(compressed_comm);
+        let comm = std::option::extract(&mut comm);
+
+        (
+            RangeProof { bytes },
+            comm
+        )
+    }
+
+    #[test_only]
+    /// Computes a range proof for a commitment to `val` with randomness `r`.
+    /// The commitment is of the form `val * val_base + r * rand_base`.
+    /// Returns both the commitment and the corresponding range proof. Only works for `num_bits` in `{8, 16, 32, 64}`.
+    public fun prove_range(
```

Comment:
This was added but doesn't seem to be used.

I guess if you added it, at least change `prove_range_pedersen` to call it? Instead of repeating the same code:

```
public fun prove_range_pedersen(val: &Scalar, r: &Scalar, num_bits: u64, dst: vector<u8>): (RangeProof, pedersen::Commitment) {
  prove_range(val, r, &ristretto255::basepoint(), &ristretto255::hash_to_point_base(), num_bits, dst)
}
```

Reply:
Done